### PR TITLE
fix: add additional tests for Regular Expression challenge

### LINKED ISSUE #17819: Lack of strings to test if provided pattern cover all requirements (regex)
#### Describe your problem and - if possible - how to reproduce it
One of the RegEX challenge(share the link) specifies 3 rules. But there are not about string match test to cover/test all rules(rules 1, and 3)
Rule no.1 as following

> 1) The only numbers in the username have to be at the end. There can be zero or more of them at the end.

And Rule no.3 specifies as following:

> 3) Usernames have to be at least two characters long. A two-letter username can only use alphabet letter characters.

Correct me if I'm wrong, but it asks the camper, the RegEx pattern should not pass something like `C4`(should be failed becasue of rule 3), and something like `Porsche911R`(could be failed becasue of rule 1).

Problem is, there is no any string to test patterns strictly if rules no 1 and 3 applied, like strings provided as example above.
The expected pattern could be something like `/\D{2,}\d*/i` , but also it's possible to pass all tests with a incorrect pattern like `/\D\d*/i` or worse `/ \D\w/` (check the second link)

You may either remove the Rules 1, and 3. Or bring more string match test to make sure user specified correct match.

Thanks.

#### Add a Link to the page with the problem

https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames/

https://forum.freecodecamp.org/t/regex-restrict-possible-usernames/198582

#### Tell us about your browser and operating system

* Browser Name: Chrome
* Browser Version: 67
* Operating System: Win7

### PR DESCRIPTION
**Pull request by cazyw**
Add additional tests for Rules 1 and 3 as currently some regular expression patterns pass the automated tests whilst failing to meet the requirements of all the rules.

- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] My article does not contain shortened URLs or affiliate links.

Closes #17819

### CODE REVIEW

**Comment by tbushman** [comment]
@Bouncey Tests pass locally.

**Comment by cazyw** [comment]
> > Add additional tests for Rules 1 and 3 as currently some regular expression patterns pass the automated tests whilst failing to meet the requirements of all the rules.
> > @cazyw Can you have some specific examples of the regular expressions which pass the tests, but fail the challenge requirements?
> 
> Thank you.

Hi @RandellDawson, this was based on an issue raised #17819. 
Requirements:
1. The only numbers in the username have to be at the end.
However there are no test cases given where the numbers are not at the end.

3. Usernames have to be at least two characters long. A two-letter username can only use alphabet letter characters.
There's no test cases for a two character username with a mix of alphabet letters and digits.

Example regex that pass:
`/\D{2,}\d*/` - passes existing tests but also passes A1, BadUs3rnam3
`/\D\w/` - passes existing tests but also passes A1, BadUs3rnam3

A1 is a two character long username but includes a number
BadUs3rnam3 has numbers in the middle

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ tests:
     testString: assert(!userCheck.test("007"), 'Your regex should not match <code>007</code>');
   - text: Your regex should not match <code>9</code>
     testString: assert(!userCheck.test("9"), 'Your regex should not match <code>9</code>');
+  - text: Your regex should not match <code>A1</code>
+    testString: 'assert(!userCheck.test("A1"), 'Your regex should not match <code>A1</code>');'
```
Comment (ValeraS) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md]: remove `'` at the begin and end of the line.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md [hunk 2]
```diff
@@ -35,6 +35,10 @@ tests:
...
+  - text: Your regex should not match <code>BadUs3rnam3</code>
+    testString: 'assert(!userCheck.test("BadUs3rnam3"), 'Your regex should not match <code>BadUs3rnam3</code>');'
```
Comment (ValeraS) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md]: remove `'` at the begin and end of the line.

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md (hunk 1 above)
Comment (cazyw) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md]: thanks, have removed them

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md (hunk 2 above)
Comment (cazyw) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md]: thanks, have removed them

**Comment by cazyw** [comment]
> @cazyw Could you also add a solution?

Thanks, solution added.

**Pull request merged**